runtime/interpreter/interpreter.go

```diff
@@ -2684,9 +2684,19 @@ func IsSubType(subType DynamicType, superType sema.Type) bool {
 		case *sema.VariableSizedType:
 			superTypeElementType = typedSuperType.Type
 
+			subTypeStaticType := interpreter.ConvertStaticToSemaType(typedSubType.StaticType)
+			if !sema.IsSubType(subTypeStaticType, typedSuperType) {
+				return false
+			}
```

Comment:
Once the runtime mutability checks (checking the type of array/dictionary before insertions) are added, I think we can rely only on this check, and maybe don't need to check both the static type and the dynamic type. Because that 'mutability checks on insertions' ensures that any value within the container is a subtype of the static type.